### PR TITLE
Add yml

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Add yml for Azure devops build
image

### CODE REVIEW

File: build/devops.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+resources:
+- repo: self
+  clean: true
+phases:
+
+- phase: windowsx86
+  displayName: 'Build Windows x86'
+  queue:
+    name: Hosted VS2017
+    demands:
```
Comment (anporumb) [inline build/devops.yml]: >demands [](start = 4, length = 7)

shouldn't the demands here list cmake too? Just curious

File: build/devops.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  - task: BatchScript@1
+    displayName: 'Setup VS vars'
+    inputs:
+      filename: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\Tools\VsDevCmd.bat"'
+      modifyEnvironment: true
+
+  - task: CMake@1
+    displayName: CMake
+    inputs:
+      cmakeArgs: '.. -Drun_int_tests:bool=ON -G "Visual Studio 15 2017"'
```
Comment (anporumb) [inline build/devops.yml]: >cmakeArgs [](start = 6, length = 9)

maybe args should explicitly list "Debug" as the config.

File: build/devops.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  - script: 'make -j$(nproc)'
+    workingDirectory: build
+    displayName: make
+
+  - task: CmdLine@1
+    displayName: 'Run ctest'
+    inputs:
+      filename: ctest
+      arguments: '-C "Debug" -V --output-on-failure'
```
Comment (anporumb) [inline build/devops.yml]: >arguments [](start = 6, length = 9)

can also take a -j $(nproc)

File: build/devops.yml (hunk 1 above)
Comment (dcristoloveanu) [reply build/devops.yml]: Yeah, that's an interesting one. I generated this yaml from a build edited in DevOps.
And lo and behold cmake was in demands.
Then I scheduled the build and it complained that cmake requirement can not be met.
But the agents have it and the build runs if I remove the cmake demand.

So, can't do better than this ...

---
In reply to: [270525439](https://github.com/Azure/azure-macro-utils-c/pull/7#discussion_r270525439) [](ancestors = 270525439)

File: build/devops.yml (hunk 2 above)
Comment (dcristoloveanu) [reply build/devops.yml]: This phase builds RelWithDebInfo. Do you mean to add RelWithDebInfo to the displayName?

---
In reply to: [270525694](https://github.com/Azure/azure-macro-utils-c/pull/7#discussion_r270525694) [](ancestors = 270525694)

File: build/devops.yml (hunk 3 above)
Comment (dcristoloveanu) [reply build/devops.yml]: Done.

---
In reply to: [270526264](https://github.com/Azure/azure-macro-utils-c/pull/7#discussion_r270526264) [](ancestors = 270526264)

File: build/devops.yml (hunk 2 above)
Comment (anporumb) [reply build/devops.yml]: No action is further needed, reviewer was confused!!!!

---
In reply to: [270540939](https://github.com/Azure/azure-macro-utils-c/pull/7#discussion_r270540939) [](ancestors = 270540939,270525694)

**Pull request merged**